internal/config/config.go

```diff
@@ -1,45 +1,88 @@
-	AmazonRegion                  string        `env:"AMAZON_REGION" envDefault:"us-west-2"`
-	AmazonRequestTimeout          time.Duration `env:"AMAZON_REQUEST_TIMEOUT" envDefault:"5s"`
-	AmazonConnectionTimeout       time.Duration `env:"AMAZON_CONNECTION_TIMEOUT" envDefault:"10s"`
+	PostgresqlHosts               string        `koanf:"postgresql_hosts"`
+	PostgresqlDbName              string        `koanf:"postgresql_db_name"`
+	PostgresqlUsername            string        `koanf:"postgresql_username"`
+	PostgresqlPassword            string        `koanf:"postgresql_password"`
+	PostgresqlSslMode             string        `koanf:"postgresql_ssl_mode"`
+	PostgresqlPort                string        `koanf:"postgresql_port"`
+	RedisHosts                    string        `koanf:"redis_hosts"`
+	RedisPort                     string        `koanf:"redis_port"`
+	RedisUsername                 string        `koanf:"redis_username"`
+	RedisPassword                 string        `koanf:"redis_password"`
+	RedisReadTimeout              time.Duration `koanf:"redis_read_timeout"`
+	RedisWriteTimeout             time.Duration `koanf:"redis_write_timeout"`
+	PostgresqlReadTimeout         time.Duration `koanf:"postgresql_read_timeout"`
+	PostgresqlWriteTimeout        time.Duration `koanf:"postgresql_write_timeout"`
+	InMemoryDbUpdateInterval      time.Duration `koanf:"in_memory_db_update_interval"`
+	OpenAiKey                     string        `koanf:"open_ai_key"`
+	StatsProvider                 string        `koanf:"stats_provider"`
+	AdminPass                     string        `koanf:"admin_pass"`
+	ProxyTimeout                  time.Duration `koanf:"proxy_timeout"`
+	NumberOfEventMessageConsumers int           `koanf:"number_of_event_message_consumers"`
+	OpenAiApiKey                  string        `koanf:"open_ai_api_key"`
+	CustomPolicyDetectionTimeout  time.Duration `koanf:"custom_policy_detection_timeout"`
+	AmazonRegion                  string        `koanf:"amazon_region"`
+	AmazonRequestTimeout          time.Duration `koanf:"amazon_request_timeout"`
+	AmazonConnectionTimeout       time.Duration `koanf:"amazon_connection_timeout"`
 }
 
-func ParseEnvVariables() (*Config, error) {
-	cfg := &Config{}
-	err := env.Parse(cfg)
+func prepareDotEnv(envFilePath string) error {
+	if envFilePath == "" {
+		ex, err := os.Executable()
+		if err != nil {
+			panic(err)
+		}
+		exPath := filepath.Dir(ex)
+
+		// first check .env, then .env_{DEV|TEST|UAT|PROD}
+		envFile := exPath + "/.env"
+		envFilePath = envFile
+	}
+
+	err := godotenv.Load(envFilePath)
+	return err
+}
+
+var k = koanf.New(".")
+
+func LoadConfig() (*Config, error) {
+	err := prepareDotEnv(".env")
 	if err != nil {
-		return nil, err
+		panic(err)
+	}
+
+	cfgPath := os.Getenv("CONFIG_FILE_NAME")
+	if cfgPath != "" {
+		f := file.Provider(cfgPath)
+		if err := k.Load(f, json.Parser()); err != nil {
+			log.Fatalf("error loading config from json file: %v", err)
+		}
+		k.Print()
```

Comment:
this should be removed

Reply:
the code is for the purpose of loading cfg from json file.
my purpose is, general cfg is set via json file. and db scretes is stored in env. 
please let me know if you prefer to use .env directly instead of json.

Reply:
i am talking about this line should be removed cause `k.Print()` will expose env variables which is not great for security